solr/solrj/src/test/org/apache/solr/common/util/TestJavaBinCodec.java

```diff
@@ -287,6 +289,45 @@ public void testAllTypes() throws IOException {
     );
   }
 
+  @Test
+  public void testReadMapEntry() throws IOException {
+    Map.Entry<Object, Object> extryFromTextDoc1 = getMapFromJavaBinCodec(SOLRJ_DOCS_1);
+    Map.Entry<Object, Object> extryFromTextDoc1_clone = getMapFromJavaBinCodec(SOLRJ_DOCS_1);
+
+    Map.Entry<Object, Object> extryFromTextDoc2 = getMapFromJavaBinCodec(SOLRJ_DOCS_2);
+    Map.Entry<Object, Object> extryFromTextDoc2_clone = getMapFromJavaBinCodec(SOLRJ_DOCS_2);
+
+    // exactly same document read twice should have same content
+    assert (extryFromTextDoc1.equals(extryFromTextDoc1_clone));
```

Comment:
While this would work because we force to enable assertions while running tests, we usually use JUnit's assertions, such as `assertEquals("", obj1, obj2);`

Reply:
sure !